consensus/mint-client/src/mint_config_tx_file.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use displaydoc::Display;
+use mc_consensus_service_config::{SignerIdentity, SignerIdentityError, SignerIdentityMap};
+use mc_transaction_core::{mint::MintConfigTxPrefix, TokenId};
+use serde::{Deserialize, Serialize};
+use serde_json::Error as JsonError;
+use std::{fs, io::Error as IoError, path::Path};
+
+/// A MintConfig in a human readable format. This is meant to be JSON
+/// serialized.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct MintConfig {
+    /// The maximum amount that this configuration is allowed to mint during the
+    /// time it is active.
+    pub mint_limit: u64,
+
+    /// Governors - the set of keys that can sign mint transactions.
+    pub minters: SignerIdentity,
+}
+
+/// A file format for serializing/deserializing a MintConfigTx in a human
+/// readable format. This is meant to be JSON serialized.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct MintConfigTxFile {
+    /// Token id being configured.
+    pub token_id: TokenId,
+
+    /// Mint configurations
+    pub configs: Vec<MintConfig>,
+
+    /// Nonce (64 bytes, hex-encoded), which is optional (empty array) in the
+    /// case we want this tool to auto-generate one.
+    #[serde(default, with = "hex")]
+    pub nonce: Vec<u8>,
```

Comment:
:question: Curious about the advantages `default` has over `Option`.
Does the `with = "hex"` always need a value?
Or is it that it's just as simple in the code to key off of an empty vector versus handling `None` vs `Some(V)`

Reply:
In this case it is because `with = "hex"` does not work with `Option<>`, and there doesn't appear to be an issue with using an empty vec to indicate this hasn't been provided. It also simplifies the code a little bit because there;s no need to check for both cases (`None` and `is_empty()`.